packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -903,6 +949,9 @@ export const Table = <
         ref={useMergedRefs(tableRef, (element) => {
           ownerDocument.current = element?.ownerDocument;
           resizeRef(element);
+          if (enableVirtualization) {
+            setTableElement(element);
```

Comment:
nit: this does not need to be guarded under `if`. It can be passed directly into `useMergedRefs`

```jsx
ref={useMergedRefs(tableRef, setTableElement, (element) => {
```

Reply:
I have the function being passed directly into `useMergedRefs` in the latest commit, but this change is causing some maximum update depth exceeded errors in the `Table` unit tests.
![image](https://github.com/iTwin/iTwinUI/assets/141063311/c87e0764-4428-4d6b-9477-b443e9499505)
Any ideas on how to work around this without guarding the `setTableElement` function? Should I try to replace these unit tests with e2e tests?

Reply:
These errors are a red herring and usually indicative of a larger problem somewhere else. You should try to reproduce this issue in a playground and spend some time debugging it.

Reply:
I was eventually able to reproduce this issue, but only by not memoizing the `Table` `data`. Therefore, I don't think this issue reflects a problem in the `Table` component that needs to be fixed. Reproduction code:
```tsx
import { Button, Table } from '@itwin/itwinui-react';
import { Column } from '@itwin/itwinui-react/react-table';
import React from 'react';

const App = () => {
  const [reverseColumnOrder, setReverseColumnOrder] = React.useState(false);

  type TestDataType = {
    name: string;
    description: string;
    accessor?: string;
    subRows?: TestDataType[];
    booleanValue?: boolean;
  };

  const mockColumns: Column<TestDataType>[] = [
    {
      id: 'name',
      Header: 'Name',
      accessor: 'name',
      width: 90,
    },
    {
      id: 'description',
      Header: 'Description',
      accessor: 'description',
      maxWidth: 200,
    },
    {
      id: 'view',
      Header: 'View',
      Cell: () => {
        return <span>View</span>;
      },
    },
  ];

  const mockColumnsReverse = [...mockColumns].reverse();

  const mockedData: TestDataType[] = [...new Array(3)].map((_, index) => ({
    name: `Name${index + 1}`,
    description: `Description${index + 1}`,
  }));

  return (
    <>
      <Button
        onClick={() => {
          setReverseColumnOrder(!reverseColumnOrder);
        }}
      >
        Reverse Columns
      </Button>
      <Table
        columns={reverseColumnOrder ? mockColumnsReverse : mockColumns}
        data={mockedData}
        emptyTableContent='Empty table'
        emptyFilteredTableContent='No results. Clear filter.'
        enableColumnReordering
        isSortable
      />
    </>
  );
};

export default App;

```
Investigated the failing tests and found that they were making multiple calls to a `mockedData` function that were unnecessary. Removing those extra calls fixed the issues and caused the tests to pass. Done in this [commit](https://github.com/iTwin/iTwinUI/pull/2092/commits/26b4b6e0f06e24f4bca901599550a4661bba3ff0).